src/server/routes/contract/read/read-batch.ts

```diff
@@ -0,0 +1,170 @@
+import { Type, type Static } from "@sinclair/typebox";
+import type { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import SuperJSON from "superjson";
+import {
+  encode,
+  getContract,
+  prepareContractCall,
+  readContract,
+  resolveMethod,
+} from "thirdweb";
+import { prepareMethod } from "thirdweb/contract";
+import { decodeAbiParameters } from "viem/utils";
+import type { AbiFunction } from "viem";
+import { createCustomError } from "../../../middleware/error";
+import { getChainIdFromChain } from "../../../utils/chain";
+import { standardResponseSchema } from "../../../schemas/shared-api-schemas";
+import { getChain } from "../../../../shared/utils/chain";
+import { thirdwebClient } from "../../../../shared/utils/sdk";
+import { prettifyError } from "../../../../shared/utils/error";
+
+const MULTICALL3_ADDRESS = "0xcA11bde05977b3631167028862bE2a173976CA11";
+
+const MULTICALL3_AGGREGATE_ABI =
+  "function aggregate3((address target, bool allowFailure, bytes callData)[] calls) external payable returns ((bool success, bytes returnData)[])";
+
+const readCallRequestItemSchema = Type.Object({
+  contractAddress: Type.String(),
+  functionName: Type.String(),
+  functionAbi: Type.Optional(Type.String()),
+  args: Type.Optional(Type.Array(Type.Any())),
+});
+
+const readMulticallRequestSchema = Type.Object({
+  calls: Type.Array(readCallRequestItemSchema),
+  multicallAddress: Type.Optional(
+    Type.String({
+      description: `Address of the multicall contract to use. If omitted, multicall3 contract will be used (${MULTICALL3_ADDRESS}).`,
+    }),
+  ),
+});
+
+const responseSchema = Type.Object({
+  results: Type.Array(
+    Type.Object({
+      success: Type.Boolean(),
+      result: Type.Any(),
+    }),
+  ),
+});
+
+const paramsSchema = Type.Object({
+  chain: Type.String(),
+});
+
+type RouteGeneric = {
+  Params: { chain: string };
+  Body: Static<typeof readMulticallRequestSchema>;
+  Reply: Static<typeof responseSchema>;
+};
+
+export async function readMulticallRoute(fastify: FastifyInstance) {
```

Comment:
Nit: I consider multicall the implementation, and any interface higher than this method shouldn't need to know about it. So var/fn names (even if not user facing) don't need to have multicall in them. Maybe `batchReadRoute` or `readBatchRoute`?